pkg/receive/handler.go

```diff
@@ -363,25 +395,32 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	responseStatusCode := http.StatusOK
 	err = h.handleRequest(ctx, rep, tenant, &wreq)
 	if err != nil {
 		level.Debug(tLogger).Log("msg", "failed to handle request", "err", err)
+		switch determineWriteErrorCause(err, 1) {
+		case errNotReady:
+			responseStatusCode = http.StatusServiceUnavailable
+			http.Error(w, err.Error(), responseStatusCode)
+		case errUnavailable:
+			responseStatusCode = http.StatusServiceUnavailable
+			http.Error(w, err.Error(), responseStatusCode)
+		case errConflict:
+			responseStatusCode = http.StatusConflict
+			http.Error(w, err.Error(), responseStatusCode)
+		case errBadReplica:
+			responseStatusCode = http.StatusBadRequest
+			http.Error(w, err.Error(), responseStatusCode)
+		default:
+			level.Error(tLogger).Log("err", err, "msg", "internal server error")
+			responseStatusCode = http.StatusInternalServerError
+			http.Error(w, err.Error(), responseStatusCode)
+		}
 	}
-
-	switch determineWriteErrorCause(err, 1) {
-	case nil:
-		return
-	case errNotReady:
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
-	case errUnavailable:
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
-	case errConflict:
-		http.Error(w, err.Error(), http.StatusConflict)
-	case errBadReplica:
-		http.Error(w, err.Error(), http.StatusBadRequest)
-	default:
-		level.Error(tLogger).Log("err", err, "msg", "internal server error")
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	h.writeTimeseriesTotal.WithLabelValues(strconv.Itoa(responseStatusCode), r.Method, tenant).Add(float64(len(wreq.Timeseries)))
+	for _, timeseries := range wreq.Timeseries {
+		h.writeSamplesTotal.WithLabelValues(strconv.Itoa(responseStatusCode), r.Method, tenant).Add(float64(len(timeseries.Samples)))
```

Comment:
This is interesting... so with histogram we will store number samples in each series in bucketed view.... But this will be unrelated to requests really. Which is  fine I guess, just a specific behaviour of this metric. But then metrics should be called `write_samples_per_series_total`.

Alternative is to  have `write_samples_total`  which will observe "sum" of samples in all timeseries - might be more reasonable to  have. cc @moadz as it's similar to T-shirt sizes request discussion - What happened to that initiative? Did we add anything at the end?

Reply:
It was supposed to be the sum of samples in all timeseries since the start. I'll fix it and keep an eye on @moadz answer 👀 👍

Reply:
The PR for query metrics has not been raised yet.

Reply:
ack. Started some yolo discussions about limits though (semi-related https://github.com/thanos-io/thanos/issues/5438)